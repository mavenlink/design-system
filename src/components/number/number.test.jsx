import React, { createRef } from 'react';
import {
  render,
  screen,
  cleanup,
} from '@testing-library/react';
import Number from './number.jsx';

describe('Number', () => {
  const requiredProps = {
    id: 'test-component',
    label: 'Test Component',
  };

  afterEach(cleanup);

  it('has defaults', () => {
    const ref = createRef();
    render(<Number {...requiredProps} ref={ref} />);
    expect(document.body).toMatchSnapshot();
    expect(ref.current).toMatchSnapshot();
  });

  describe('label', () => {
    it('presents a label', () => {
      render(<Number {...requiredProps} />);
      expect(screen.getByLabelText('Test Component')).toBeInTheDocument();
    });
  });
});
