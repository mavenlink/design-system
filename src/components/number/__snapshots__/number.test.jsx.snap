// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Number has defaults 1`] = `
<body>
  <div>
    <div
      role="presentation"
    >
      <label
        class="label"
        for="test-component"
      >
        Test Component
      </label>
      <div
        class="control-container"
      >
        <input
          aria-describedby="test-componentHint"
          id="test-component"
          type="number"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`Number has defaults 2`] = `null`;
